x-pack/legacy/plugins/ingest/server/libs/adapters/policy/default.ts

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { PathReporter } from 'io-ts/lib/PathReporter';
+import { SODatabaseAdapter } from '../so_database/default';
+import { RuntimePolicyFile, NewPolicyFile } from './adapter_types';
+
+import { PolicyFile, DatasourceInput, BackupPolicyFile } from './adapter_types';
+
+export class PolicyAdapter {
+  constructor(private readonly so: SODatabaseAdapter) {}
+
+  public async create(
+    policy: NewPolicyFile
+  ): Promise<{ id: string; shared_id: string; version: number }> {
+    const newSo = await this.so.create<PolicyFile>('policies', (policy as any) as PolicyFile);
+
+    return {
+      id: newSo.id,
+      shared_id: newSo.attributes.shared_id,
+      version: newSo.attributes.version,
+    };
+  }
+
+  public async get(id: string): Promise<PolicyFile> {
+    const policySO = await this.so.get<PolicyFile>('policies', id);
+
+    if (policySO.error) {
+      throw new Error(policySO.error.message);
+    }
+
+    if (!policySO.attributes) {
```

Comment:
From what I saw implenting so adapters, so is going to throw a boom error if there is no saved object

Reply:
Good catch. I had added this when debugging an issue (mistaking the bug for the need for the conditional) and never removed it post-bug-fix.